### PR TITLE
feature/Reactnative-add-bottomsheet

### PR DESCRIPTION
**Pull request by NikhilBehari**
Changes proposed in this Pull Request:
- Add bottomsheet functionality and tests for React Native framework

### CODE REVIEW

**Comment by jeffvg** [comment]
- Font weight between ionic and react-native slightly different
- React native bottom sheet has one divider and ionic has two dividers

File: src/Alarms.tsx [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+        bottom: 0,
+        right: 0,
+        width: '100%',
+        maxWidth: 600,
+    },
+    bottomSheetItemTitle: {
+        paddingLeft: 16,
+    },
+    listHeader: {
+        fontSize: 17,
```
Comment (joebochill) [inline src/Alarms.tsx]: We shouldn't be manually setting font sizes (use the Typography elements)

File: src/Alarms.tsx [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+                                iconColor={item.active ? Colors.white[100] : Colors.black[500]}
+                                fontColor={item.active ? Colors.red[500] : Colors.black[500]}
+                                statusColor={item.active ? Colors.red[500] : Colors.white[100]}
+                                avatar={item.active}
+                            />
+                        ))}
+                    </ScrollView>
+                </SafeAreaView>
+                {this.state.showBottomSheet ? (
+                    <View
```
Comment (joebochill) [inline src/Alarms.tsx]: Do we need to manually create this overlay element? Can't we just use the Modal component from React-Native for the bottomsheet?

File: src/Alarms.tsx [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+                        style={styles.overlay}
+                        onTouchStart={(): void => this.setState({ showBottomSheet: false })}
+                        testID={'overlay'}
+                    ></View>
+                ) : null}
+                <BottomSheet style={styles.footer} show={this.state.showBottomSheet}>
+                    <View style={styles.rowHeader}>
+                        <Text style={styles.listHeader}>Sort By: </Text>
```
Comment (joebochill) [inline src/Alarms.tsx]: Use Typography components instead of Text

File: src/Alarms.tsx [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+                                active={this.state.showSessions}
+                                label={'Sessions'}
+                                onPress={(): void => this.setState({ showSessions: !this.state.showSessions })}
+                            >
+                                {' '}
+                            </IconToggle>
+                        </View>
+                    </View>
+                    <Divider />
+                    <ListItem
```
Comment (joebochill) [inline src/Alarms.tsx]: Should this be InfoListItem, or is the ListItem style close enough to ours?

File: src/components/IconToggle.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        margin: 20,
+    },
+});
+
+class IconToggle extends React.PureComponent<IconToggleProps> {
+    constructor(props: IconToggleProps) {
+        super(props);
+    }
+
+    render(): JSX.Element {
```
Comment (joebochill) [inline src/components/IconToggle.tsx]: Is this an opportunity for us to use the Hero component?

File: src/Alarms.tsx (hunk 3 above)
Comment (joebochill) [reply src/Alarms.tsx]: This is important because if you use a regular Text component, the font will be wrong

File: src/Alarms.tsx (hunk 2 above)
Comment (huayunh) [reply src/Alarms.tsx]: Yes. I wondered the exact same thing when I was working on [the bottomsheet RN example](https://github.com/pxblue/bottomsheet/blob/reactnative/src/BottomSheet.tsx), and I tried sniffing around to see if there were any material ui examples. Looks like we have to do this :P

Comment (joebochill) [reply src/Alarms.tsx]: https://facebook.github.io/react-native/docs/modal
https://github.com/react-native-community/react-native-modal

File: src/Alarms.tsx (hunk 4 above)
Comment (huayunh) [reply src/Alarms.tsx]: `InfoListItem` will display a right chevron when `onPress` is set, which we don't want in this case. While I can fool it by doing `rightComponent={<View></View>}`, I guess it is not a good solution...

File: src/components/IconToggle.tsx (hunk 5 above)
Comment (huayunh) [reply src/components/IconToggle.tsx]: Using Hero:
image

Using View:
image

I don't like how the rest parts of the Channel Value is truncated for 'Active Alarm'. But if I change the fontSize to 'tiny' then it looks like this: 
image

File: src/Alarms.tsx (hunk 2 above)
Comment (huayunh) [reply src/Alarms.tsx]: Now I couldn't have everything inside safe area :(

File: src/components/IconToggle.tsx (hunk 5 above)
Comment (huayunh) [reply src/components/IconToggle.tsx]: Notes after discussion with Joseph: ended up sticking with View but added in flexWrap for better display on smaller screens

File: src/Alarms.tsx (hunk 4 above)
Comment (huayunh) [reply src/Alarms.tsx]: Notes after discussion with Joseph: InfoListItem would truncate tall line heights, which makes the appearance not as satisfactory as that of ListItem. Going with ListItem but with fonts, colors and spacings adjusted.

File: src/Alarms.tsx (hunk 2 above)
Comment (huayunh) [reply src/Alarms.tsx]: Notes after discussion with Joseph: ended up using [@react-native-community/react-native-safe-area-view](https://github.com/react-native-community/react-native-safe-area-view) with forceInset

**Pull request merged**